sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/TaskUtils.java

```diff
@@ -271,37 +224,33 @@ public static boolean volumesEqual(TaskSpec first, TaskSpec second) {
     }
 
     /**
-     * Returns a path=>template mapping of the provided {@link ConfigFileSpec}s. Assumes
-     * that each config file is given a distinct path.
+     * Returns a name=>template mapping of the provided {@link ConfigFileSpec}s. Checks for unique paths and names
+     * across all configs.
      *
      * @throws IllegalArgumentException if multiple config specifications have matching relative path strings
      */
-    private static Map<String, String> getConfigTemplateMap(
-            Collection<ConfigFileSpec> configSpecifications) throws IllegalArgumentException {
-        Map<String, String> configMap = new HashMap<>();
-        for (ConfigFileSpec configSpecification : configSpecifications) {
-            String prevValue =
-                    configMap.put(configSpecification.getRelativePath(), configSpecification.getTemplateContent());
-            if (prevValue != null) {
+    private static Map<String, ConfigFileSpec> getConfigTemplateMap(Collection<ConfigFileSpec> configSpecs)
+            throws IllegalArgumentException {
+        Set<String> configPaths = new HashSet<>();
+        Map<String, ConfigFileSpec> configMap = new HashMap<>();
+        for (ConfigFileSpec configSpec : configSpecs) {
+            if (!configPaths.add(configSpec.getRelativePath())) {
+                throw new IllegalArgumentException(String.format(
```

Comment:
Yes. `TaskUtilsTest` has both `testConfigsSamePathFailsValidation` and `testConfigsSameNameFailsValidation`